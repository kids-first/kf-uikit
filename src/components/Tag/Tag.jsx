import React from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';

/**
 * A Simple rounded button
 */
const Tag = ({ size, color, className, children }) => {

  const buttonClass = classes(
    'Tag',
    {[`Tag--${size}`]: ['small', 'large'].includes(size)},
    `Tag--${color}`,
    className
  );
  return (
    <span className={buttonClass}>
      {children}
    </span>
  );
}

Tag.propTypes = {
  /** The size of the button */
  size: propTypes.string,
  /** The color of the button. */
  color: propTypes.string,
  /** Any additional classes to be applied to the button */
  className: propTypes.string,
  /** Children elements. */
  children: propTypes.string,
};

Tag.defaultProps = {
  size: 'default',
  color: 'primary',
  className: null,
  children: null,
};

export default Tag;
