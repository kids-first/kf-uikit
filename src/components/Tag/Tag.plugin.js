/* eslint-disable import/no-extraneous-dependencies */
const _ = require('lodash')
const defaultConfig = require('tailwindcss/defaultConfig')()

function defaultOptions() {
  return {
    borderRadius: '1.5rem',
    fontWeight: '300',
    lineHeight: '1.0',
    fontSize: '0.8rem',
    padding: '.4rem 0.8rem',
    colors: {
      white: {
        background: defaultConfig.colors.white,
        text: defaultConfig.colors.black,
      },
      black: {
        background: defaultConfig.colors.black,
        text: defaultConfig.colors.white,
      },
    },
    sizes: {
      small: {
        fontSize: '.6rem',
        padding: '.3rem .6rem',
        borderRadius: '1.0rem',
      },
      large: {
        fontSize: '1.25rem',
        padding: '.75rem 1.5rem',
        borderRadius: '1.7rem',
      }
    }
  }
}

module.exports = (opts) => {
  const options = _.isFunction(opts)
   ? opts(defaultOptions())
   : _.defaults(opts, defaultOptions())

  return ({ addComponents, e }) => {
    addComponents([
      {
        '.Tag': {
          display: 'inline-block',
          padding: options.padding,
          fontSize: options.fontSize,
          fontWeight: options.fontWeight,
          lineHeight: options.lineHeight,
          borderRadius: options.borderRadius,
          textDecoration: 'none',
        }
      },
      ..._.map(_.omit(options.sizes, 'default'), (sizeOptions, name) => ({
        [`.Tag--${e(name)}`]: {
          padding: _.get(sizeOptions, 'padding', options.padding),
          fontSize: _.get(sizeOptions, 'fontSize', options.fontSize),
          fontWeight: _.get(sizeOptions, 'fontWeight', options.fontWeight),
          lineHeight: _.get(sizeOptions, 'lineHeight', options.lineHeight),
          borderRadius: _.get(sizeOptions, 'borderRadius', options.borderRadius),
        }
      })),
      ..._.map(options.colors, (colorOptions, name) => ({
        [`.Tag--${e(name)}`]: {
          border: `2px solid ${colorOptions.background}`,
          backgroundColor: colorOptions.background,
          color: colorOptions.text,
        }
      }))
    ])
  }
}
