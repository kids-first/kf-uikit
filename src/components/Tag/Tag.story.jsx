/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { select, text } from '@storybook/addon-knobs';
import Tag from './Tag';

const stories = storiesOf('Tags', module);

stories.add('Simple Tag', () => (
  <Tag
    size={select('Size', ['small', 'default', 'large'], 'default')}
    color={select('Color', ['primary', 'secondary', 'tertiary', 'error', 'grey'], 'grey')}
  >
    {text('Text', '10,567')}
  </Tag>
  ),
  {
		info: {
      text: `
        Used to show a numeric stat or icon.
        `
    },
  }
)
