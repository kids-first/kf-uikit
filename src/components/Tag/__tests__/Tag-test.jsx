import React from 'react';
import renderer from 'react-test-renderer';
import Tag from '../Tag';

['default', 'small', 'large'].map((size) => {
  [true, false].map((outline) => {
    ['primary', 'secondary', 'tertiary', 'error', 'grey'].map((c) => {
      it(`${size} ${c} badge renders correctly`, () => {
          const tree = renderer.create(
                  <Tag
                    color={c}
                    size={size}
                  >
                    Hello
                  </Tag>
                  ).toJSON();
            expect(tree).toMatchSnapshot();
      });
    });
  });
});
