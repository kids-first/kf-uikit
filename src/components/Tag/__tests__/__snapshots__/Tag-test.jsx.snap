// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default error badge renders correctly 1`] = `
<span
  className="Tag Tag--error"
>
  Hello
</span>
`;

exports[`default error badge renders correctly 2`] = `
<span
  className="Tag Tag--error"
>
  Hello
</span>
`;

exports[`default grey badge renders correctly 1`] = `
<span
  className="Tag Tag--grey"
>
  Hello
</span>
`;

exports[`default grey badge renders correctly 2`] = `
<span
  className="Tag Tag--grey"
>
  Hello
</span>
`;

exports[`default primary badge renders correctly 1`] = `
<span
  className="Tag Tag--primary"
>
  Hello
</span>
`;

exports[`default primary badge renders correctly 2`] = `
<span
  className="Tag Tag--primary"
>
  Hello
</span>
`;

exports[`default secondary badge renders correctly 1`] = `
<span
  className="Tag Tag--secondary"
>
  Hello
</span>
`;

exports[`default secondary badge renders correctly 2`] = `
<span
  className="Tag Tag--secondary"
>
  Hello
</span>
`;

exports[`default tertiary badge renders correctly 1`] = `
<span
  className="Tag Tag--tertiary"
>
  Hello
</span>
`;

exports[`default tertiary badge renders correctly 2`] = `
<span
  className="Tag Tag--tertiary"
>
  Hello
</span>
`;

exports[`large error badge renders correctly 1`] = `
<span
  className="Tag Tag--large Tag--error"
>
  Hello
</span>
`;

exports[`large error badge renders correctly 2`] = `
<span
  className="Tag Tag--large Tag--error"
>
  Hello
</span>
`;

exports[`large grey badge renders correctly 1`] = `
<span
  className="Tag Tag--large Tag--grey"
>
  Hello
</span>
`;

exports[`large grey badge renders correctly 2`] = `
<span
  className="Tag Tag--large Tag--grey"
>
  Hello
</span>
`;

exports[`large primary badge renders correctly 1`] = `
<span
  className="Tag Tag--large Tag--primary"
>
  Hello
</span>
`;

exports[`large primary badge renders correctly 2`] = `
<span
  className="Tag Tag--large Tag--primary"
>
  Hello
</span>
`;

exports[`large secondary badge renders correctly 1`] = `
<span
  className="Tag Tag--large Tag--secondary"
>
  Hello
</span>
`;

exports[`large secondary badge renders correctly 2`] = `
<span
  className="Tag Tag--large Tag--secondary"
>
  Hello
</span>
`;

exports[`large tertiary badge renders correctly 1`] = `
<span
  className="Tag Tag--large Tag--tertiary"
>
  Hello
</span>
`;

exports[`large tertiary badge renders correctly 2`] = `
<span
  className="Tag Tag--large Tag--tertiary"
>
  Hello
</span>
`;

exports[`small error badge renders correctly 1`] = `
<span
  className="Tag Tag--small Tag--error"
>
  Hello
</span>
`;

exports[`small error badge renders correctly 2`] = `
<span
  className="Tag Tag--small Tag--error"
>
  Hello
</span>
`;

exports[`small grey badge renders correctly 1`] = `
<span
  className="Tag Tag--small Tag--grey"
>
  Hello
</span>
`;

exports[`small grey badge renders correctly 2`] = `
<span
  className="Tag Tag--small Tag--grey"
>
  Hello
</span>
`;

exports[`small primary badge renders correctly 1`] = `
<span
  className="Tag Tag--small Tag--primary"
>
  Hello
</span>
`;

exports[`small primary badge renders correctly 2`] = `
<span
  className="Tag Tag--small Tag--primary"
>
  Hello
</span>
`;

exports[`small secondary badge renders correctly 1`] = `
<span
  className="Tag Tag--small Tag--secondary"
>
  Hello
</span>
`;

exports[`small secondary badge renders correctly 2`] = `
<span
  className="Tag Tag--small Tag--secondary"
>
  Hello
</span>
`;

exports[`small tertiary badge renders correctly 1`] = `
<span
  className="Tag Tag--small Tag--tertiary"
>
  Hello
</span>
`;

exports[`small tertiary badge renders correctly 2`] = `
<span
  className="Tag Tag--small Tag--tertiary"
>
  Hello
</span>
`;
